nav2_rviz_plugins/src/nav2_panel.cpp

```diff
@@ -845,6 +859,14 @@ Nav2Panel::onInitialize()
         navigation_feedback_indicator_->setText(getNavThroughPosesFeedbackLabel());
       }
     });
+  follow_waypoints_goal_status_sub_ = node->create_subscription<action_msgs::msg::GoalStatusArray>(
```

Comment:
This might be where you duplication lies? I'm not sure why this is necessary for the status of the WPF action

Reply:
We still need this so that when the WPF action finishes, the elapsed time field gets cleaned up being reset to 0. Without this section, we would need to infer that it is finished from another action, which I have seen is NOT always true. Perhaps it's a bug, but I didn't want to assume it was the same. 

Also, I'm trying to find the duplication for why the status is rendered twice, but nothing I am seeing is calling that part of the code again in my diff.

Reply:
OK that's fine but you should remove the UX element of it beyond just resetting to zero (if this even is where that duplication is created)

Reply:
I'll split these up. Because the data comes from two different callbacks, and you also want it displayed uniformly (a single table), I think this means I'd have to cache the data from the two feedbacks as a member variable, then refactor the formatter to conditionally output parts of the stored data depending on the state.

Reply:
@padhupradheep Can you help me by sharing the motivation for keeping the Nav through poses and waypoint following as the same state? Any strong opposition to making these separate states in `nav2_panel`? 

Steve has asked, reasonably, that the UI information for waypoints only shows up when navigating through waypoints. I think that makes sense. Aside from adding some flags to keep track of if the waypoint navigation was started, cancelled, or finished, I was thinking splitting that stuff into another state would be a good approach. Your input is appreciated.

Reply:
They should be the same for gathering the waypoints, but not the same during processing - they should diverge in the state machine as there are 2 different buttons (NavThroughPose vs WPF) which trigger that transition.